### PR TITLE
docs: New APM configuration section

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR is mostly structural. It moves the `Configure` section up two levels in the navigation. This requires many heading-level changes. I also moved all of the `configure` documentation to a new subdirectory so it's easier to find and edit.

Separating this into its own PR should make the content changes that are coming easier to review.

### CODE REVIEW

File: docs/configure/general.asciidoc [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+|====
+
+Maximum request header::
+Maximum permitted size of a request's header accepted by the server to be processed (in Bytes).
++
+|====
+| APM Server binary | `apm-server.max_header_size`
+| Fleet-managed     | `Maximum size of a request's header`
+|====
+
```
Comment (bmorelli25) [inline docs/configure/general.asciidoc:36]: This is what the new configure documentation will look like.

**Review by simitt — APPROVED** [review]
I have several comments for the content, but keep them (mostly) for the dedicated content PR. From the structural changes, this LGTM

File: docs/configure/general.asciidoc [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+Default: `127.0.0.1:8200`
++
+|====
+| APM Server binary | `apm-server.host`
+| Fleet-managed     | `Host`
+|====
+
+URL::
+The publicly reachable server URL. For deployments on Elastic Cloud or ECK, the default is unchangeable.
++
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/configure/general.asciidoc:23]: As discussed offline with Silvia on 10/05/23, I'll address this and all other feedback in the follow-up PR.

File: docs/configure/instrumentation.asciidoc [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-== Configure APM instrumentation
+= Configure APM instrumentation
 
 ++++
 <titleabbrev>Instrumentation</titleabbrev>
 ++++
 
-Libbeat uses the Elastic APM Go Agent to instrument its publishing pipeline.
-Currently, only the {es} output is instrumented.
+NOTE: Instrumentation of APM Server is not supported for Fleet-managed APM.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/configure/instrumentation.asciidoc:8]: Except for this one because it's easy!

**Pull request merged**